.github/workflows/reusable-maven.yml

```
@@ -0,0 +1,98 @@
+name: Reusable workflow for maven packages
+
+on:
+  workflow_call:
+    inputs:
+      path_to_package:
+        required: true
+        type: string
+      dependencies:
+        description: "Array of paths to dependencies"
+        required: false
+        type: string
+
+defaults:
+  run:
+    working-directory: java
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      - name: Build dependencies
+        if: ${{ inputs.dependencies != '' }}
+        run: |
+          IFS=',' read -r -a dirs <<< "${{ inputs.dependencies }}"
+          for dir in "${dirs[@]}"
+          do
+            cd "$dir"
+            mvn clean install --no-transfer-progress
+            cd -
+          done
+
+      - name: Verify package
+        run: |
+          cd ${{ inputs.path_to_package }}
+          mvn verify
```

I would add two flags here:
`-fae`: Fail at the end in order to run all tests and projects in case the package has children. I do not think it's strictly necessary but also doesn't hurt.
`--no-transfer-progress`: avoid spamming the logs

```suggestion
          mvn verify -fae --no-transfer-progress
```